src/sentry/rules/history/preview.py

```diff
@@ -198,36 +200,66 @@ def get_fired_groups(
 
 
 def get_top_groups(
-    project: Project, start: datetime, end: datetime, condition_activity: GroupActivityMap
+    project: Project,
+    start: datetime,
+    end: datetime,
+    condition_activity: GroupActivityMap,
+    dataset_map: Dict[str, Dataset],
 ) -> GroupActivityMap:
     """
     Filters the activity to contain only groups that have the most events in the past 2 weeks.
 
     Since frequency conditions require one snuba query per groups, we need to limit the number groups we process.
     """
-    if not condition_activity:
-        return condition_activity
-    # TODO: Also check other datasets for top groups
-    groups = raw_query(
-        dataset=Dataset.Events,
-        start=start,
-        end=end,
-        filter_keys={"project_id": [project.id]},
-        conditions=[("group_id", "IN", list(condition_activity.keys()))],
-        aggregations=[("count", "group_id", "groupCount")],
-        groupby=["group_id"],
-        order_by="-groupCount",
-        seleted_columns=["group_id"],
-        limit=FREQUENCY_CONDITION_GROUP_LIMIT,
-    ).get("data", [])
-
-    top_groups = {group["group_id"] for group in groups}
+    datasets = {dataset_map.get(group) for group in condition_activity.keys()}
+
+    # queries each dataset for top x groups and then gets top x overall
```

Comment:
i don't necessarily love this approach (and the post query sorting) but i need to think of a better alternative solution

Reply:
any way we can use `bulk_raw_query` instead? it's not used anywhere else I can find in the codebase, thought. not sure what the frontend does for search either, maybe ping @barkbarkimashark

Reply:
Yeah, if you're querying multiple datasets you can get some performance benefits by using `bulk_raw_query` to query snuba in parallel. See:https://github.com/getsentry/sentry/blob/8079be388f76a49dfefdc36c19d2f566af2b9376/src/sentry/search/snuba/executors.py#L263

It basically builds up multiple queries according to the GroupCategory and queries snuba in bulk in a threadpool.

Reply:
If you're planning to support alert previews for other GroupCategories besides error-issues or performance issues, it might make sense to introduce a similar pattern like in `snuba_search`

Reply:
As you can see here: https://github.com/getsentry/sentry/blob/7d3c6d49d983178170e7bc0deacbafe73dbc5f95/src/sentry/utils/snuba.py#L657-L688
`raw_query` already makes use of `bulk_raw_query` under the hood. As a first step, you can replace your current usage of `raw_query` with `bulk_raw_query` but still iterate over the datasets to test out the correct behavior before trying out the parallel stuff.

Reply:
thanks! lemme try this

Reply:
I know of at least one other place where bulk query should be used, but I'll include that in another pr

Reply:
> If you're planning to support alert previews for other GroupCategories besides error-issues or performance issues, it might make sense to introduce a similar pattern like in `snuba_search`

Yeah this is a good idea, I'll put this in a diff pr